arduino-ide-extension/src/node/clang-formatter.ts

```diff
@@ -80,8 +80,11 @@ export class ClangFormatter implements Formatter {
   }
 
   private async dataDirPath(): Promise<string> {
-    const { dataDirUri } = await this.configService.getConfiguration();
-    return FileUri.fsPath(dataDirUri);
+    const { config, messages } = await this.configService.config();
+    if (!config) {
+      throw new Error(messages ? messages[0] : 'cli config was undefined');
```

Comment:
caller should handle undefined, formatter must still work without a valid data dir

Reply:
https://github.com/arduino/arduino-ide/pull/1774/commits/30c4af958cab9ffb3672a6c26bd13d6771a95291